components/eventing-controller/pkg/handlers/jetstream_test.go

```diff
@@ -1199,3 +1202,170 @@ func setupTestEnvironment(t *testing.T) *TestEnvironment {
+		{
+			name: "should be equal if the two instances are identical",
+			givenIdentifier1: NewSubscriptionSubjectIdentifier(
+				evtesting.NewSubscription("sub-1", "ns-1"),
+				"prefix.app.event.operation.v1",
+			),
+			givenIdentifier2: NewSubscriptionSubjectIdentifier(
+				evtesting.NewSubscription("sub-1", "ns-1"),
+				"prefix.app.event.operation.v1",
+			),
+			wantEqual: true,
+		},
+		// instances are not equal
+		{
+			name: "should not be equal if only name is different",
+			givenIdentifier1: NewSubscriptionSubjectIdentifier(
+				evtesting.NewSubscription("sub-1", "ns-1"),
+				"prefix.app.event.operation.v1",
+			),
+			givenIdentifier2: NewSubscriptionSubjectIdentifier(
+				evtesting.NewSubscription("sub-2", "ns-1"),
+				"prefix.app.event.operation.v1",
+			),
+			wantEqual: false,
+		},
+		{
+			name: "should not be equal if only namespace is different",
+			givenIdentifier1: NewSubscriptionSubjectIdentifier(
+				evtesting.NewSubscription("sub-1", "ns-1"),
+				"prefix.app.event.operation.v1",
+			),
+			givenIdentifier2: NewSubscriptionSubjectIdentifier(
+				evtesting.NewSubscription("sub-1", "ns-2"),
+				"prefix.app.event.operation.v1",
+			),
+			wantEqual: false,
+		},
+		{
+			name: "should not be equal if only subject is different",
+			givenIdentifier1: NewSubscriptionSubjectIdentifier(
+				evtesting.NewSubscription("sub-1", "ns-1"),
+				"prefix.app.event.operation.v1",
+			),
+			givenIdentifier2: NewSubscriptionSubjectIdentifier(
+				evtesting.NewSubscription("sub-1", "ns-1"),
+				"prefix.app.event.operation.v2",
+			),
+			wantEqual: false,
+		},
+		// possible naming collisions
+		{
+			name: "should not be equal if subject is the same but name and namespace are swapped",
+			givenIdentifier1: NewSubscriptionSubjectIdentifier(
+				evtesting.NewSubscription("sub-1", "ns-1"),
+				"prefix.app.event.operation.v1",
+			),
+			givenIdentifier2: NewSubscriptionSubjectIdentifier(
+				evtesting.NewSubscription("ns-1", "sub-1"),
+				"prefix.app.event.operation.v1",
+			),
+			wantEqual: false,
+		},
+		{
+			name: "should not be equal if subject is the same but name and namespace are only equal if joined together",
+			givenIdentifier1: NewSubscriptionSubjectIdentifier(
+				evtesting.NewSubscription("sub-1", "ns-1"), // evaluates to "sub-1ns-1" when joined
+				"prefix.app.event.operation.v1",
+			),
+			givenIdentifier2: NewSubscriptionSubjectIdentifier(
+				evtesting.NewSubscription("sub-1n", "s-1"), // evaluates to "sub-1ns-1" when joined
+				"prefix.app.event.operation.v1",
+			),
+			wantEqual: false,
+		},
+	}
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			gotInstanceEqual := reflect.DeepEqual(tc.givenIdentifier1, tc.givenIdentifier2)
+			assert.Equal(t, tc.wantEqual, gotInstanceEqual)
+
+			gotConsumerNameEqual := tc.givenIdentifier1.ConsumerName() == tc.givenIdentifier2.ConsumerName()
+			assert.Equal(t, tc.wantEqual, gotConsumerNameEqual)
```

Comment:
(nit) is `require.equal()` preferable?

Reply:
As discussed offline, we'll keep it like that because `t.Run` would run all test cases anyway and `require.Equal` would not have the desired effect here.